clang/lib/AST/DeclTemplate.cpp

```diff
@@ -309,35 +309,35 @@ bool TemplateDecl::isTypeAlias() const {
 void RedeclarableTemplateDecl::anchor() {}
 
 RedeclarableTemplateDecl::CommonBase *RedeclarableTemplateDecl::getCommonPtr() const {
-  if (Common)
-    return Common;
+  if (CommonBase *C = Common.getPointer())
```

Comment:
Looks like we should have a function on `RedeclarableTemplateDecl` to do this... its a shame that `getCommonPtr` is taken, but perhaps a private helper here would be nice.  Something like `getCommonPtrInternal` or something?  With a comment that makes it clear what it is doing.